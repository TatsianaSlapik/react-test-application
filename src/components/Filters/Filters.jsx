import React, { useState } from "react";

export default function Filters() {
  return (
    <div className="filters">
      <header className="filters__header">
        <h2 className="filters__header__title">Filters</h2>
      </header>
      <div className="filters__inner">
        <div className="filters__filter">
          <h3 className="filters__filter__title">Category</h3>
          <ul>
            <li>
              <input type="checkbox" />
              <label>1</label>
            </li>
            <li>
              <input type="checkbox" />
              <label>2</label>
            </li>
            <li>
              <input type="checkbox" />
              <label>3</label>
            </li>
            <li>
              <input type="checkbox" />
              <label>4</label>
            </li>
          </ul>
        </div>
        <div className="filters_price">
          <h3 className="filters__filter__title">Price</h3>
          <p>
            <input maxlength="25" size="20" />
            -
            <input maxlength="25" size="20" />
          </p>
        </div>
      </div>
    </div>
  );
}
