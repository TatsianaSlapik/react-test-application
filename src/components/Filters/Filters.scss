.filters {
  overflow: auto;
  width: 25%;
  flex-shrink: 0;
  @media screen and (max-width: 320px) {
    width: 100%;
  }
}

.filters_inner {
  margin-left: 36px;
}

.filters_inner_li {
  list-style: none;
  margin-block: 21px;
  input {
    transform: scale(1.8);
  }
  label {
    font-size: 21px;
    margin-left: 16px;
  }
}

.filters_price_inner {
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  align-items: baseline;

  margin-block: 21px;

  input {
    transform: scale(2.8);
    @media screen and (max-width: 320px) {
      transform: scale(1.8);
    }
  }
}
