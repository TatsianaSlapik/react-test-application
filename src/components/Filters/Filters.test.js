import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

import Filters from "./Filters.jsx";
const mockStore = configureStore([]);

describe("Filters", () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({
      productsList: [],
      listViewState: false,
    });
    component = renderer.create(
      <Provider store={store}>
        <Filters />
      </Provider>
    );
  });

  it("Filters", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
