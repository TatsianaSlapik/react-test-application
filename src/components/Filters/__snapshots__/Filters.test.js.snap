// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters Filters 1`] = `
<div
  className="filters"
>
  <div
    className="filters_inner"
  >
    <div
      className="accordion-item__line"
      onClick={[Function]}
    >
      <h3
        className="filters_filter_title"
      >
        Category
      </h3>
      <span
        className="accordion-item_icon"
      />
    </div>
    <div
      className="accordion_content"
    >
      <ul>
        <li
          className="filters_inner_li"
        >
          <input
            type="checkbox"
          />
          <label>
            Smartphones
          </label>
        </li>
        <li
          className="filters_inner_li"
        >
          <input
            type="checkbox"
          />
          <label>
            iPhone
          </label>
        </li>
      </ul>
    </div>
    <div
      className="accordion-item__line"
      onClick={[Function]}
    >
      <h3
        className="filters_filter_title"
      >
        Price
      </h3>
      <span
        className="accordion-item_icon"
      />
    </div>
    <div
      className="accordion_content"
    >
      <div
        className="filters_price_inner"
      >
        <input
          maxLength="25"
          size="5"
        />
        â€•
        <input
          maxLength="25"
          size="5"
        />
      </div>
    </div>
  </div>
</div>
`;
